lib/asset_cloud.rb

```diff
@@ -42,12 +42,18 @@ def execute_callbacks(symbol, args)
     @extensions.each {|ext| ext.execute_callbacks(symbol, args)}
   end
 
+  protected
+
+  def valid_key_path?(key)
```

Comment:
Why are you passing the key as an argument? This is just checking the class's own `key`